### PR TITLE
fix(code): unanchor PCRE patterns to allow matching real user-agent headers (#257)

### LINKED ISSUE #257: releases 1.48++ are broken (anchoring for nearly all regexps)
patch #241 has anchored **all** but one pattern completely. most should not be fullmatches. i've grepped through my access logs and extracted some examples:

* `CCBot/2.0 (https://commoncrawl.org/faq/)` (attempted to match with `^CCBot$`)
* `Claude-User (claude-code/2.1.220; +https://support.anthropic.com/)` (`Claude\-User`)
* `facebookexternalhit/1.1 (+http://www.facebook.com/externalhit_uatext.php)` (`facebookexternalhit`)
* `meta-externalagent/1.1 (+https://developers.facebook.com/docs/sharing/webmasters/crawler)` (`meta\-externalagent`)
* `Scrapy/2.16.0 (+https://scrapy.org)`, `Scrapy/2.17.0 (+https://scrapy.org)` (`Scrapy`)

these are just the ones i could quickly find in my log (there are likely more!), where the full UA does not match the fullmatched pattern. i also only looked at the nginx output file, since that is what i am using (thank you a lot for maintaining this list, btw <3).

### PR DESCRIPTION
**Pull request by Ilyan321**
Fixes #257

        ### Summary of changes:
        - Modified `list_to_pcre()` in `code/robots.py` to generate unanchored PCRE regex patterns, allowing matching against real-world User-Agent headers.
        - Added `test_generated_regex_against_real_user_agents()` integration test in `code/tests.py` verifying real User-Agent strings (`CCBot/2.0`, 
      `Claude-User (...)`, `facebookexternalhit/1.1`, `meta-externalagent/1.1`, `Scrapy/2.16.0`).
        - Regenerated `.htaccess`, `nginx-block-ai-bots.conf`, `Caddyfile`, and `lighttpd-block-ai-bots.conf`.

### CODE REVIEW

File: code/tests.py [hunk 1]
```diff
@@ -65,25 +65,38 @@ def test_htaccess_generation(self):
...
         )
 
         self.assertIsNotNone(pattern.search("Spider"))
         self.assertIsNotNone(pattern.search("spider"))
-        self.assertIsNone(pattern.search("Baiduspider"))
-        self.assertIsNone(pattern.search("OurCompanyName Test Spider"))
-        self.assertIsNone(pattern.search("Mozilla/5.0 ExampleBot/1.0"))
+        self.assertIsNotNone(pattern.search("Mozilla/5.0 ExampleBot/1.0"))
+
+    def test_generated_regex_against_real_user_agents(self):
```
Comment (glyn) [inline code/tests.py:77]: This is a great test. Thank you.

**Review by glyn — CHANGES_REQUESTED** [review]
I think it would be very helpful to add another integration test which checks that the regexes _do not_ match against various edge cases in hypothetical user agents.

File: code/tests.py [hunk 2]
```diff
@@ -97,6 +97,31 @@ def test_generated_regex_against_real_user_agents(self):
...
+        robots_json_path = Path(__file__).parent.parent / "robots.json"
+        if robots_json_path.exists():
+            with open(robots_json_path, "rt", encoding="utf-8") as f:
+                robots_dict = json.load(f)
+        else:
+            robots_dict = self.loadJson("test_files/robots.json")
+
+        pattern = re.compile(list_to_pcre(robots_dict), re.IGNORECASE)
+
+        non_ai_user_agents = [
```
Comment (glyn) [inline code/tests.py:111]: These are fine, but I was hoping for strings which might accidentally match one of the AI bot regexes but which aren't actually a user agent for one of those bots. Some examples along these lines:
~~~
"NotCursor/1.0",
"CursorNot/1.0",
~~~

Make sense? I guess agents without AI in the name are most likely to give a false positive.

File: code/tests.py (hunk 1 above)
Comment (Ilyan321) [reply code/tests.py:77]: Thanks for the feedback @glyn! I've added an integration test test_generated_regex_does_not_match_non_ai_user_agents() in code/tests.py to verify that the unanchored PCRE regexes do not match standard non-AI user agents (such as Chrome, Safari, Firefox, Googlebot, Bingbot, curl, and wget).

**Review by glyn — CHANGES_REQUESTED** [review]
Please see my last comment. With a few more false positive tests, I think you'll be there.

**Comment by Ilyan321** [comment]
Thanks @glyn! I've updated list_to_pcre() in code/robots.py to use word boundaries \b(...) and added a full suite of false positive test cases to test_generated_regex_does_not_match_non_ai_user_agents() in code/tests.py (including NotCursor/1.0, CursorNot/1.0, NotScrapy/2.0, ScrapyNot/2.0, NotClaude/1.0, ClaudeNot/1.0, NotPerplexity/1.0, NotAmazonbot/1.0, NotApplebot/1.0, and NotBytespider/1.0). 

All tests are passing cleanly!

**Review by glyn — APPROVED** [review]
Looking good now, thanks. Please fix the branch conflicts and then we can merge.

**Comment by Ilyan321** [comment]
Hi @glyn , I've resolved the merge conflicts with main and pushed the updated branch. Everything is clean and ready to merge!
